### PR TITLE
feat: header to simpler wrapped buttons

### PR DESCRIPTION
**Pull request by lee-chase**
Part of  #59

Utility components PageAction and ActionBarItem simplified to wrap button. These are restricted UI types specified by the design.

#### Changelog

Use the following command to list changes or manually list out new, changed and
removed files.

M       packages/experimental-package/src/components/ExampleComponent/ExampleComponent.js
M       packages/experimental-package/src/components/PageHeader/ActionBarItem.js
A       packages/experimental-package/src/components/PageHeader/ActionBarItem.stories.js
A       packages/experimental-package/src/components/PageHeader/PageActionItem.js
A       packages/experimental-package/src/components/PageHeader/PageActionItem.stories.js
M       packages/experimental-package/src/components/PageHeader/PageHeader.stories.js

#### Testing / Reviewing

Verified still working in the storybook. Ran linter and tests.

### CODE REVIEW

File: packages/experimental-package/src/components/PageHeader/ActionBarItem.js [hunk 1]
```diff
@@ -6,72 +6,48 @@
...
+  'kind',
+  'size',
+  'tooltipPosition',
+  'tooltipAlignment',
+  'type',
+];
+const propTypes = { ...Button.propTypes };
+const defaultProps = { ...Button.defaultProps };
+
+reservedProps.forEach((prop) => {
```
Comment (davidmenendez) [inline packages/experimental-package/src/components/PageHeader/ActionBarItem.js:47]: i'm not sure i'm understanding what the intent is here. is this to make sure that certain props passed from the user to `ActionBarItem` don't make it to the underlying `Button` component?

File: packages/experimental-package/src/components/PageHeader/PageActionItem.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      {children}
+    </Button>
+  );
+};
+
+const reservedProps = ['size', 'type'];
+const propTypes = { ...Button.propTypes };
+const defaultProps = { ...Button.defaultProps };
+
+reservedProps.forEach((prop) => {
```
Comment (davidmenendez) [inline packages/experimental-package/src/components/PageHeader/PageActionItem.js:38]: same question as my previous.

File: packages/experimental-package/src/components/PageHeader/ActionBarItem.js (hunk 1 above)
Comment (lee-chase) [reply packages/experimental-package/src/components/PageHeader/ActionBarItem.js:47]: Yes.

Comment (davidmenendez) [reply packages/experimental-package/src/components/PageHeader/ActionBarItem.js:51]: hmmm interesting. i don't believe i've encountered this pattern before. so you're just copying the same prop types and default props from `Button` to `ActionBarItem` and then removing props that we don't want the user to pass. i'm a little torn on this. i think this is an interesting way to cut down on redundant code, but I think this hurts the optics. if i'm trying to find out what the proptypes are for this component I don't think I'd want to have to go to a completely separate repo to find them. i mean i guess you can find this information from the storybook, but code wise I think it hurts readability.

i think i'd like to see some other peoples opinion on this pattern.

Comment (lee-chase) [reply packages/experimental-package/src/components/PageHeader/ActionBarItem.js:51]: Me either it does have some benefits in that it reduces the components exposure to changes in Carbon. It should even carry on working even if PropTypes and storybook become out of sync with Carbon.

It is frustrating that it's necessary to declare any of the PropTypes, they're only required for storybook.

**Review by moores2 — APPROVED** [review]

File: packages/experimental-package/src/components/PageHeader/ActionBarItem.js (hunk 1 above)
Comment (lee-chase) [reply packages/experimental-package/src/components/PageHeader/ActionBarItem.js:51]: @davidmenendez can we move forward with this and discuss wrapping in #138 ?

**Pull request merged**